### ISSUE TITLE
[bitnami/elasticsearch] No processor type exists with name [inference]

### ISSUE THREAD

**Issue by alimoezzi** [bitnami/containers#72437]
### Name and Version

bitnami/elasticsearch:8.15.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

[Inference](https://www.elastic.co/guide/en/elasticsearch/reference/current/inference-processor.html) processor although it's available in kibana but upon creating an ingest pipeline with inference processor results in the following error:
```
No processor type exists with name [inference] 
```
![image](https://github.com/user-attachments/assets/cea16334-ae9a-473b-b313-f666e3d62ec0)

This processor works with no problem with docker.elastic.co/elasticsearch/elasticsearch:8.15.0

### What is the expected behavior?

Bitnami build should include this processor as the official image also contains it both in kibana and elasticsearch.

### What do you see instead?

Kibana includes the option but upon saving this error is raised:
```
No processor type exists with name [inference]
```

**Comment by andresbono** [comment]
Hi @alimoezzi, thank you for opening this issue. Can you describe and share how are you running the elasticsearch stack? Are you using the Helm Chart? Do you run the containers directly? Please, give us some more details so we can try to reproduce the issue on our side.

**Comment by alimoezzi** [comment]
I have tested both containers and helm chart and none of them works.

**Comment by alimoezzi** [comment]
To reproduce the issue, you need to create a pipeline with inference Processor. Also, this processor is missing in the cluster info.

**Comment by alimoezzi** [comment]
I think it is related to disabled ML by default in both container and helm chart.
I tried to enable it but then the cluster doesn't turn on. In this case I think for sure Bitnami build is the cause as official build works with no problem.

**Comment by gongomgra** [comment]
Hi @alimoezzi,

Sorry for the delay on getting back to you. Can you provide us with more information on how to reproduce the issue and how to enable the ML features so we can test it on our side? According to the official docs linked below you have to download and install your own ML model into the stack first and then create the inference type processor using the installed model. 

https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest-pipeline-search-inference.html

Did you find any error following those steps? I have found the "Inference" processor type under the "Stack Management" > "Ingest pipelines" > "Create pipeline" > "Add a processor" menu

![Screenshot 2024-10-11 at 17 07 29](https://github.com/user-attachments/assets/c01176c7-8ea2-4566-b68e-bfd7e8f186c4)

**Comment by alimoezzi** [comment]
> I have found the "Inference" processor type under the "Stack Management" > "Ingest pipelines" > "Create pipeline" > "Add a processor" menu

Exactly, at this stage continue by clicking 'Add Processor' and then create the pipeline! Then you will get the same error.

**Comment by alimoezzi** [comment]
This feature is not enabled by **adding a model**. Instead by enabling feature flag `elasticsearch_conf_set xpack.ml.enabled "true"` as it has been disable by Bitnami by default without any notice.

**Comment by gongomgra** [comment]
@alimoezzi thanks for the detailed information. I have found a related configuration in our initialization logic

https://github.com/bitnami/containers/blob/2de4377adaaa5ba0ba682bc1aa62747cea63bc22/bitnami/elasticsearch/7/debian-12/rootfs/opt/bitnami/scripts/libelasticsearch.sh#L753

According to it, we found some issues while enabling it by default in some scenarios, but you can customize the setting on your side via an initialization script

https://github.com/bitnami/containers/tree/main/bitnami/elasticsearch#initializing-a-new-instance

**Comment by alimoezzi** [comment]
The bigger problem is bitnami build has corrupted xpack-ml. Please enable the option and you see the instance throws error.

**Comment by gongomgra** [comment]
@alimoezzi thanks for your message. I have created an internal task to further investigate this issue. We will keep you posted.
